online/quiz_scraper.py

```diff
@@ -9,67 +9,92 @@
+    (points to correct option).
+
+    Args:
+        number (:obj:`int`): The number of question(s) (with respective choices and answers) to return. Must be between
+            1 and 10 (both included).
+    """
+
+    def remover():
+        """Helper function to remove respective question, choice and answer."""
+        try:
+            all_questions.remove(question)
+            all_choices.remove(choices)
+            all_answers.remove(answer)
+        except ValueError:
+            pass
 
     page = r.randint(1, 76)
-    logger(message=f"Quiz obtained from {page=}.", debug=True)
+    logger(message=f"Quiz obtained from {page=}.")
     quiz_url = f"{QUIZ_URL}/{page}"
-    content = requests.get(quiz_url).content
 
-    soup = BeautifulSoup(content, 'html.parser')
-    results = soup.find_all(class_=r.choice(['even', 'odd']))
+    soup = BeautifulSoup(requests.get(quiz_url).content, 'lxml')
+    results_ques = soup.find_all(class_=lambda a_class: a_class == 'odd' or a_class == 'even')
+
     all_questions = []
     all_choices = []  # Type: List[List[str(choices)]]
     all_answers = []
 
-    for result in results:
+    for result in results_ques:
+        ops = result.find_all(['br', 'sup', 'sub'])
         question_choices = []
 
-        question = result.find('b')  # Get question
-        question = question.text.strip()
-        question = question[3:]  # Removes the 'Q. ' part
-
-        if len(question) > 255:  # If we've reached max question character limit for ptb
-            return
-
-        all_questions.append(question)
-
-        options = result.find_all('br')  # Get options as a list
-
-        if len(options) > 5:  # If there are notes or something weird (cheap way out, but practicality >>)
-            options = options[:5]
-
-        for option in options:  # Get string between <br> tags by printing next sibling
-            choice = option.nextSibling
-
-            if choice is None:  # Edge case when there's that stupid note
-                text = option.text.strip()
-                spliced = re.sub('([0-4][)])*', '', text).split()
-                question_choices.extend(spliced)
+        if len(ops) == 5:
+            ops.pop()
+
+        if result.b.string is not None:
+            question = str(result.b.string)
+        else:
+            if result.i is not None:  # If there are italics in question
+                question = str(result.b.get_text())  # Get text without italics
+            else:
+                question = str(result.b.next_element.string)
+
+        all_questions.append(question[3:].strip())  # [3:] is to remove 'Q. '
+
+        option = ''  # String to add choices
+        for op in ops:  # Loop through choices if extra formatting such as subscripts or superscripts are present
+
+            sub_sup = re.sub('([0-9][)] )*', '', op.next_element.string).strip()  # Remove choice numbers like '1 )'
+            if op.name == 'sup':  # If superscript present, add that
```

Comment:
```suggestion
            if op.name in ['sup', 'sub']:  # In case of superscripts/subscripts
```

combine the `if` and the `elif` into one

Reply:
No the substitution is different

Reply:
Ah sorry I missed that